src/robot/robot.ts

```diff
@@ -146,4 +148,15 @@ export interface Robot {
     resourceNames?: ResourceName.AsObject[],
     durationMs?: number
   ): RobotStatusStream;
+
+  /**
+   * Call a function when an event of either 'reconnected' or 'disconnected' is 
+   * triggered.
+   * 
+   * @param event - The event that was triggered.
```

Comment:
[nit] we might want to note the event types here.

Separately, we might also want to call out that these events only apply to WebRTC connections.